e2e-tests/src/test/kick_out.rs

```diff
@@ -0,0 +1,158 @@
+use aleph_client::{
+    change_validators, get_committee_kick_out_config, get_current_era_validators,
+    get_current_session, get_kick_out_reason_for_validator, get_next_era_non_reserved_validators,
+    get_next_era_reserved_validators, get_underperformed_validator_session_count,
+    wait_for_at_least_session, wait_for_event, wait_for_full_era_completion, AccountId,
+    SignedConnection, XtStatus,
+};
+use codec::Decode;
+use log::info;
+use primitives::{
+    CommitteeSeats, KickOutReason, SessionCount, DEFAULT_CLEAN_SESSION_COUNTER_DELAY,
+    DEFAULT_KICK_OUT_MINIMAL_EXPECTED_PERFORMANCE, DEFAULT_KICK_OUT_SESSION_COUNT_THRESHOLD,
+};
+
+use crate::{
+    accounts::{account_ids_from_keys, get_validator_seed, NodeKeys},
+    rewards::set_invalid_keys_for_validator,
+    validators::get_test_validators,
+    Config,
+};
+
+const VALIDATOR_TO_DISABLE_NON_RESERVED_INDEX: u32 = 0;
+const VALIDATOR_TO_DISABLE_OVERALL_INDEX: u32 = 2;
+const NODE_TO_DISABLE_ADDRESS: &str = "127.0.0.1:9945";
+const SESSIONS_TO_MEET_KICK_OUT_THRESHOLD: SessionCount = 4;
+
+/// Runs a chain, sets up a committee and validators. Sets an incorrect key for one of the
+/// validators. Waits for the offending validator to hit the kick-out threshold of sessions without
+/// producing blocks. Verifies that the offending validator has in fact been kicked out for the
+/// appropriate reason.
+pub fn kick_out_automatic(config: &Config) -> anyhow::Result<()> {
+    let root_connection = config.create_root_connection();
+
+    let validator_keys = get_test_validators(config);
+    let reserved_validators = account_ids_from_keys(&validator_keys.reserved);
+    let non_reserved_validators = account_ids_from_keys(&validator_keys.non_reserved);
+
+    let seats = CommitteeSeats {
+        reserved_seats: 2,
```

Comment:
Could you please add a task to put variants of this test to nightly in the future. Those should be _parametrized_ e2e tests, for which we don't have yet support - a test case like this would be `run_kick_out_automatic(seats: CommitteeSeats, validatosr_to_disable:: &[AccountId])`

Reply:
Sure, added: https://cardinal-cryptography.atlassian.net/browse/A0-1495